resources/experimental-feature.md

```diff
@@ -0,0 +1,89 @@
+However, the experimental feature process can be an opportunity to test an improvement (for example, regarding performance) we are not comfortable putting immediately into production. In this case, more bugs and/or API changes are expected.
+
+## Communication with users
+
+For each experimental feature:
+- In the [product repository](https://github.com/meilisearch/product/discussions)
+  - A dedicated GitHub discussion is open
+  - The discussion should be clear about the following information
+      - Our motivations: why this feature is unstable?
+      - How to use the feature: how to activate it? Do we need to do a migration with a dump?
+      - Our future plan: what do we expect to make it stable? When will the feature potentially be stable?
+  - The discussion should not stay in a stale state. Users need to understand they can interact in the discussion and get quick answers. If needed, the discussion will be updated regularly.
+- Using the Meilisearch CLI: the `--help` command in the Meilisearch binary will redirect to the related GitHub discussion and warn about the unstable state of the features.
+- In the documentation: a small dedicated page should be done about the purpose of the experimental feature and redirect to the related GitHub discussion.
+
+## Some warnings about the usage
+
+- Experimental features are not recommended to be used in production.
+- For some provided experimental features, re-indexing will be needed. For this, more than only restarting the search engine, using a dump will be needed to enforce re-indexing.
+This means, in some situations, using a dump might be needed to activate an experimental feature but also to deactivate it. The users will be clearly informed in the related discussion how to activate the feature
+
+> ⚠️ When deactivating this kind of experimental feature, since our process is not mature yet, users can have some issues with their DB, even when using a dump.<br>
+> We recommend users always save their data (with snapshots and/or dumps) before starting to activate experimental features. Of course, the safest method is to use a script to feed Meilisearch.
+
+## Technical explanations
+
+### Why does Meilisearch need to be restarted (by using the `experimental` CLI flag)?
+
+Meilisearch uses LMDB to store the data, not only user documents but also a lot of internal data, like Meilisearch tasks. With LMDB, if we want to add or remove data structures to store this data, we need to close the LMDB environment and re-open it, so to restart Meilisearch.
+
+It means if an experimental feature implementation involves a modification of the data structures, users must restart Meilisearch; we cannot make it enable/disable from the HTTP routes.
+
+Unfortunately, this might concern most of the future features. That’s why we chose, for the moment, only to suggest flag CLI in order to enable/disable experimental features. However, this can be changed in the future or adapted to the situation.
```

Comment:
```suggestion
### Why does Meilisearch need to be restarted when activating an experimental feature?

Meilisearch uses LMDB to store both documents and internal application data, such as Meilisearch tasks. Altering these internal data structures requires closing and re-opening the LMDB environment.

If an experimental feature implementation involves a modification of internal data structures, users must restart Meilisearch. This cannot be done via HTTP routes.

Unfortunately, this might impact most experimental features. However, this might change in the future, or adapted to the context of a specific new feature.
```